src/pages/Profile.js

```diff
@@ -58,6 +59,107 @@ const styles = {
 }
 
 
+const NihLink = ({ nihToken }) => {
+  const loadNihStatus = async () => {
+    const status = await Ajax().User.getNihStatus()
+    setNihStatus(status)
+  }
+
+  const linkNihAccount = async nihToken => {
+    setLinking(true)
+    const status = await Ajax().User.linkNihAccount(nihToken)
+    setNihStatus(status)
+    setLinking(false)
+  }
+
+
+  /*
+   * Hooks
+   */
+  const [{ linkedNihUsername, linkExpireTime, datasetPermissions }, setNihStatus] = useState({})
+  const [linking, setLinking] = useState(false)
+
+  Utils.useOnMount(() => {
+    if (nihToken) {
+      // Clear the query string, but use replace so the back button doesn't take the user back to the token
+      Nav.history.replace({ search: '' })
+      linkNihAccount(nihToken)
+    } else {
+      loadNihStatus()
+    }
+  })
+
+
+  /*
+   * Render helpers
+   */
+  const makeAccountLinkLink = label => {
+    const nihRedirectUrl = `${window.location.protocol}//${window.location.host}/#profile?nih-username-token={token}`
+
+    return link({
+      href: `${getConfig().shibbolethUrlRoot}/link-nih-account?redirect-url=${encodeURIComponent(nihRedirectUrl)}`,
+      style: { display: 'flex', alignItems: 'center' },
+      target: '_blank'
+    }, [
+      label,
+      icon('pop-out', { size: 12, style: { marginLeft: '0.5rem' } })
+    ])
+  }
+
+  const makeDatasetAuthStatus = ({ name, authorized }) => {
+    return div({ key: `nih-auth-status-${name}`, style: { display: 'flex' } }, [
+      div({ style: { flex: 1 } }, [`${name} Authorization`]),
+      div({ style: { flex: 2 } }, [
+        authorized ? 'Authorized' : 'Not Authorized',
+        !authorized && h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+          'Your account was linked, but you are not authorized to view this controlled dataset. Please go ',
+          link({
+            href: 'https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login',
+            target: '_blank'
+          }, [
+            'here',
+            icon('pop-out', { size: 12 })
+          ]),
+          ' to check your credentials.'
+        ])
+      ])
+    ])
+  }
+
+
+  /*
+   * Render
+   */
+  return h(Fragment, [
+    div({ style: styles.form.title }, [
+      'NIH Account',
+      h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+        'Linking with eRA Commons will allow FireCloud to automatically determine if you can access controlled datasets hosted in FireCloud (ex. TCGA) based on your valid dbGaP applications.'
```

Comment:
Terra, not FireCloud 
```suggestion
        'Linking with eRA Commons will allow Terra to automatically determine if you can access controlled datasets hosted in Terra (ex. TCGA) based on your valid dbGaP applications.'
```